### PR TITLE
#1583 Abstracted Refund Modal

### LINKED ISSUE #1583: Finances Fixes - Edit Refund Modal
Blocked By #1582 (you can start on this one if you want, but you won't be able to hook up the backend til this ticket is done)

### Description

Currently once a user submits a refund, they can't edit it. This means if they make a type they can never correct it. We wanna add the ability for a user to edit a refund after they have submitted it, to ensure that all the refunds saved in Finishline are accurate. This ticket is to add the frontend portion. In the 'My Refunds' section of the finance dashboard, a user should be able to click on a refund and it'll pop up the same modal for submitting a refund prefilled with whatever the existing values for the refund are.

### Acceptance Criteria

- [ ] When a user clicks on a refund in the 'My Refunds' section of the finance dashboard, it should pop open a modal to edit the refund
- [ ] This will be the same modal used for creating a refund, prefilled with the values of the refund
- [ ] Submitting this modal will send the saved values to the newly created edit reimbursement endpoint

### Proposed Solution
You should make a RefundModal component and extend it to create 'Create Refund' and 'Edit Refund' modal components. See how we do it for creating/editing an expense type for reference.

### PR DESCRIPTION
**Pull request by walker-sean**
## Changes

Made an abstract refund modal that is used by the report refund modal and created an edit refund modal that also uses it

## Screenshots

_If you made UI changes you must post a screenshot of the whole page for each change in 1) a normal sized window and 2) the smallest possible window_

_If you did any manual testing (e.g., with Postman), put screenshots of the http request and before and after of the db_

_If none of this applies, you can delete this section_

image
image

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1583

### CODE REVIEW

File: src/frontend/src/pages/FinancePage/RefundsSection.tsx [hunk 1]
```diff
@@ -141,77 +145,88 @@ const Refunds = ({ userReimbursementRequests, allReimbursementRequests }: Refund
...
-              ))}
-            </TableBody>
-          </Table>
-        </TableContainer>
+              </TableHead>
+              <TableBody>
+                {rows.map((row, index) => (
+                  <TableRow
+                    key={`${row.date}-$${row.amount}-${index}`}
+                    sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
```
Comment (Aaryan1203) [inline src/frontend/src/pages/FinancePage/RefundsSection.tsx]: Can you make the cursor a pointer here

File: src/frontend/src/pages/FinancePage/FinanceComponents/RefundModal.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      return parts[1].length <= 2;
+    })
+    .typeError('Refund amount is required'),
+  dateReceived: yup.date().required()
+});
+
+interface RefundModalProps {
+  showModal: boolean;
+  handleClose: () => void;
+  mutateAsync: (data: any) => void; // if anyone can figure out how to change this without raising type errors, be my guest
```
Comment (Aaryan1203) [inline src/frontend/src/pages/FinancePage/FinanceComponents/RefundModal.tsx]: what did you want to change it to? Like whats wrong with this

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
One small comment and a question but otherwise looks good

File: src/frontend/src/pages/FinancePage/FinanceComponents/RefundModal.tsx (hunk 2 above)
Comment (walker-sean) [reply src/frontend/src/pages/FinancePage/FinanceComponents/RefundModal.tsx]: Just that it uses `any`, but I think I know how to fix it now

**Review by Aaryan1203 — DISMISSED** [review]
Just some merge conflicts then lgtm

**Pull request merged**